### PR TITLE
itest: save temp miner's logs

### PR DESCRIPTION
**Pull request by yyforyongyu**
Current build has a test flake in `open_channel_reorg_test`, logs can be found [here](https://travis-ci.org/github/lightningnetwork/lnd/jobs/731354058). A timeout happened when setting up the temp miner.

To further assist the debugging, the temp miner's logs are saved.

### CODE REVIEW

File: lntest/itest/lnd_test.go [hunk 1]
```diff
@@ -2339,28 +2339,24 @@ func testOpenChannelAfterReorg(net *lntest.NetworkHarness, t *harnessTest) {
...
-		"--rejectnonstd",
-		"--txindex",
-		"--nowinservice",
-		"--nobanning",
-	}
-	tempMiner, err := rpctest.New(
-		harnessNetParams, &rpcclient.NotificationHandlers{}, args,
+	tempLogDir := "./.tempminerlogs"
+	tempMiner, tempMinerCleanUp, err := lntest.NewMiner(
+		tempLogDir, "output_temp_btcd_miner.log",
```
Comment (halseth) [inline lntest/itest/lnd_test.go:2344]: maybe the name of the test should be part of the file name here?

**Review by halseth — APPROVED** [review]
Thanks for the PR, nice cleanup!

File: lntest/node.go [hunk 2]
```diff
@@ -351,6 +353,57 @@ func newNode(cfg NodeConfig) (*HarnessNode, error) {
...
+			return fmt.Errorf("unable to copy file: %v", err)
+		}
+		if err = os.RemoveAll(logDir); err != nil {
+			return fmt.Errorf(
+				"cannot remove dir %s: %v", logDir, err,
+			)
+		}
+		return nil
+	}
+	return miner, cleanUp, nil
```
Comment (bhandras) [inline lntest/node.go:404]: style nit: could add a new line before return (also in the cleanup).

File: lntest/node.go [hunk 3]
```diff
@@ -351,6 +353,57 @@ func newNode(cfg NodeConfig) (*HarnessNode, error) {
...
+		// file before deleting the temporary log dir.
+		logFile := fmt.Sprintf(
+			"%s/%s/btcd.log", logDir, netParams.Name,
+		)
+		copyPath := fmt.Sprintf("./%s", logFilename)
+		err := CopyFile(copyPath, logFile)
+		if err != nil {
+			return fmt.Errorf("unable to copy file: %v", err)
+		}
+		if err = os.RemoveAll(logDir); err != nil {
```
Comment (bhandras) [inline lntest/node.go:397]: nit: maybe add a new line here too

**Review by bhandras — APPROVED** [review]
LGTM, just nits. Thanks @yyforyongyu :+1:

File: lntest/node.go (hunk 2 above)
Comment (yyforyongyu) [reply lntest/node.go:404]: fixed.

File: lntest/node.go (hunk 3 above)
Comment (yyforyongyu) [reply lntest/node.go:397]: fixed.

File: lntest/itest/lnd_test.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/itest/lnd_test.go:2344]: Yes I think that's more indicative. Changed.

**Pull request merged**